<launch>
  <arg name="index"/>
  <group ns="cars">
    <group ns="controllers">
      <group ns="lane">
        <param name="type"   value="pid"/>
        <param name="p"      value="4"/>
        <param name="ref"    value="laneOffset"/>
        <param name="sensor" value="lane"/>
        <param name="act"    value="steer"/>
      </group>
      <group ns="cruise">
        <param name="type" value="assign"/>
        <param name="ref"  value="speed"/>
        <param name="act"  value="speed"/>
      </group>
      <group ns="straight">
        <param name="type" value="assign"/>
        <param name="ref"  value="null"/>
        <param name="act"  value="steer"/>
      </group>
      <group ns="acc">
        <param name="type" value="pid"/>
        <param name="ref"  value="speed"/>
        <param name="act"  value="speed"/>
      </group>
    </group>
    <group ns="sensors">
      <group ns="lane">
        <param name="type"   value="laneSensor"/>
        <param name="handle" value="RoadSensor"/>
      </group>
      <group ns="frontDist">
        <param name="type"   value="proxDistance"/>
        <param name="handle" value="FrontDistSens"/>
      </group>
    </group>
    <group ns="actuators">
      <group ns="steer">
        <param name="type"   value="float"/>
        <param name="value"   value="0.0f"/>
      </group>
      <group ns="speed">
        <param name="type"   value="float"/>
        <param name="value"  value="0.0f"/>
      </group>
    </group>
    <group ns="references">
      <group ns="laneOffset">
        <param name="type" value="float"/>
        <param name="value" value="0.0"/>
      </group>
      <group ns="speed">
        <param name="type" value="float"/>
        <param name="value" value="30.0"/>
      </group>
      <group ns="null">
        <param name="type" value="float"/>
        <param name="value" value="0.0"/>
      </group>
    </group>
    <node name="car$(arg index)" pkg="aseia_car_sim" type="car" launch-prefix="perf stat">
      <param name="index"      value="$(arg index)"/>
      <param name="ctrls"      value="lane, cruise"/>
      <param name="sens"       value="lane, frontDist"/>
      <param name="acts"       value="speed, steer"/>
      <param name="refs"       value="laneOffset, speed"/>
    </node>
  </group>
</launch>
